### ISSUE TITLE
Cannot use static arrays as indices for views

### ISSUE THREAD

**Issue by aaortizb** [JuliaArrays/StaticArrays.jl#587]
I have a code that uses StaticArrays. It works fine in Julia 1.0.3, but the same code does not work in Julia 1.1.0. The issue is triggered when doing the following operation:

`Kg[rows,rows] .+= Ke`

where Ke is assembled on Kg. Here Kg[rows,rows]  has the same dimension of Ke, but it is a standard Julia array. On the other hand, Ke is an Static Array. Any idea on how to resolve this issue?

The error thrown is the following:

`ERROR: LoadError: ArgumentError: unable to check bounds for indices of type StaticArrays.StaticIndexing{StaticArrays.MArray{Tuple{2},Int64,1,2}}
Stacktrace:
 [1] checkindex(::Type{Bool}, ::Base.OneTo{Int64}, ::StaticArrays.StaticIndexing{StaticArrays.MArray{Tuple{2},Int64,1,2}}) at .\abstractarray.jl:504
 [2] checkbounds_indices at .\abstractarray.jl:475 [inlined]
 [3] checkbounds at .\abstractarray.jl:428 [inlined]
 [4] checkbounds at .\abstractarray.jl:449 [inlined]
 [5] view(::Array{Float64,2}, ::StaticArrays.MArray{Tuple{2},Int64,1,2}, ::StaticArrays.MArray{Tuple{2},Int64,1,2}) at .\subarray.jl:155
 [6] maybeview(::Array{Float64,2}, ::StaticArrays.MArray{Tuple{2},Int64,1,2}, ::Vararg{StaticArrays.MArray{Tuple{2},Int64,1,2},N} where N) at .\views.jl:124
 [7] dotview at .\broadcast.jl:1067 [inlined]
 [8] _dense_loop_over_elements!(::Array{Float64,2}, ::Array{Float64,1}, ::Array{Any,1}, ::Int64, ::Function, ::Function) at C:\Dropbox\Universidad\codes\Julia\GaLerKia\GaLerKia-0.1\src\FEMlia\FEMLiaLinElast\FEMLiaLinElastSolvers.jl:137
 [9] f_linear_elastostatics_dense_solver!(::Dict{String,Any}) at C:\Dropbox\Universidad\codes\Julia\GaLerKia\GaLerKia-0.1\src\FEMlia\FEMLiaLinElast\FEMLiaLinElastSolvers.jl:68
 [10] galerkia_solve!(::String, ::Dict{String,Any}) at C:\Dropbox\Universidad\codes\Julia\GaLerKia\GaLerKia-0.1\src\GalerkiaSolve.jl:11
 [11] top-level scope at none:0
in expression starting at C:\Dropbox\Universidad\codes\Julia\GaLerKia\GaLerKia-0.1\test\femlia_test001_linelast_line1d_multistory_building.jl:93`

**Comment by mbauman** [comment]
The simpler reproduction isn't dependent upon Julia versions — StaticArrays just needs to define this checkindex method:

```julia
julia> Kg = zeros(10,10)
       rows = @MArray[1,2]
       @view Kg[rows,rows]
ERROR: ArgumentError: unable to check bounds for indices of type StaticArrays.StaticIndexing{MArray{Tuple{2},Int64,1,2}}
Stacktrace:
 [1] checkindex(::Type{Bool}, ::Base.OneTo{Int64}, ::StaticArrays.StaticIndexing{MArray{Tuple{2},Int64,1,2}}) at ./abstractarray.jl:504
 [2] checkbounds_indices at ./abstractarray.jl:475 [inlined]
 [3] checkbounds at ./abstractarray.jl:428 [inlined]
 [4] checkbounds at ./abstractarray.jl:449 [inlined]
 [5] view(::Array{Float64,2}, ::MArray{Tuple{2},Int64,1,2}, ::MArray{Tuple{2},Int64,1,2}) at ./subarray.jl:155
 [6] top-level scope at none:0
```

**Comment by mbauman** [comment]
You could have different versions of StaticArrays installed on each — what's your `Pkg.status()` report?  I'm on `v0.10.2` on both.

**Comment by aaortizb** [comment]
Perhaps that is the problem. In Julia 1.0.3 I have v0.10.0 and in Julia 1.1.0, v0.10.2

**Comment by mbauman** [comment]
Aha, yup, that's it.  `StaticIndexing` was recently introduced and is just missing this method.

**Comment by andyferris** [comment]
@mbauman should we do something to resolve this?

**Comment by tribut** [comment]
This is fixed in #586, would appreciate a review.

**Comment by jbrea** [comment]
It seems to me that #586 fixes the issue only partially, e.g.
```julia
julia> X = randn(3, 10);

julia> view(X, SArray{Tuple{1}}([1]))
1-element view(::Array{Float64,1}, [1]) with eltype Float64:
 0.9175896614250181
```
but
```julia
julia> view(view(X, :, 1), [1])
1-element view(::Array{Float64,2}, [1], 1) with eltype Float64:
 0.9175896614250181

julia> view(view(X, :, 1), SArray{Tuple{1}}([1]))
Error showing value of type SubArray{Float64,0,Array{Float64,2},Tuple{StaticArrays.StaticIndexing{SArray{Tuple{1},Int64,1,1}},StaticArrays.StaticIndexing{Int64}},false}:
ERROR: MethodError: no method matching _indices_sub(::SubArray{Float64,0,Array{Float64,2},Tuple{StaticArrays.StaticIndexing{SArray{Tuple{1},Int64,1,1}},StaticArrays.StaticIndexing{Int64}},false}, ::StaticArrays.StaticIndexing{SArray{Tuple{1},Int64,1,1}}, ::StaticArrays.StaticIndexing{Int64})
Closest candidates are:
  _indices_sub(::SubArray, ::Real, ::Any...) at subarray.jl:399
  _indices_sub(::SubArray, ::AbstractArray, ::Any...) at subarray.jl:401
  _indices_sub(::SubArray) at subarray.jl:398
Stacktrace:
 [1] axes at ./subarray.jl:397 [inlined]
 [2] summary(::IOContext{REPL.Terminals.TTYTerminal}, ::SubArray{Float64,0,Array{Float64,2},Tuple{StaticArrays.StaticIndexing{SArray{Tuple{1},Int64,1,1}},StaticArrays.StaticIndexing{Int64}},false}) at ./show.jl:1837
 [3] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::SubArray{Float64,0,Array{Float64,2},Tuple{StaticArrays.StaticIndexing{SArray{Tuple{1},Int64,1,1}},StaticArrays.StaticIndexing{Int64}},false}) at ./arrayshow.jl:316
 [4] (::getfield(REPL, Symbol("#do_respond#38")){Bool,getfield(REPL, Symbol("##48#57")){REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /build/julia/src/julia-1.1.0/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:714
```
